privacy_evaluator/output/user_output.py

```diff
@@ -1,103 +1,43 @@
+        else:
+            for key, value in self.__dict__.items():
+                ret[key] = self._convertToListIfNeeded(value)
+        return json.dumps(ret)
 
-    """User Output Class"""
-
-    def __init__(
-        self,
-        attack_data_y: np.ndarray,
-        privacy_risk: np.ndarray,
-    ):
+    def _to_dict(self, filter: np.ndarray = None) -> dict:
         """
-        Initilaizes the Class with values
-        :param attack_data_y: An Array of the labels of the attck data
-        :param privacy_risk: the Privacy risk corresponding to the attack data
-        :param all_labels: All labels that are in the training set
+        output function for dicts
+        :param filter: if needed this filters the output for the given keys
         """
-        self.attack_data_y = attack_data_y
-        self.privacy_risk = privacy_risk
+        if filter is not None:
+            ret = {}
+            for key in filter:
+                ret[key] = self.__dict__.get(key)
+        return self.__dict__
 
-    def histogram_top_k(
-        self, all_labels: np.ndarray, k: int = 10, show_diagram: bool = True
-    ) -> Tuple[np.ndarray, np.ndarray]:
+    def __str__(self) -> str:
         """
-        Draw histogram of class distribution of the k points with highest privacy risk score
-        :param all_labels: all the labels of the input data
-        :param k: the number of points to consider, default 10
-        :param show_diagram: determines if the diagram should be shown, default: True
-        :return: All lables of the data with the number of points that are in the top k
+        Overwrite the String method so the output looks nicer
         """
-        sorting = np.argsort(self.privacy_risk)
-        sorting = np.flip(sorting)
-        sorted_attack_data_labels = self.attack_data_y[sorting][:k]
-        df = pd.DataFrame(
-            {
-                "labels": sorted_attack_data_labels,
-            }
-        )
-        df = df.groupby("labels")["labels"].agg(["count"])
-        labels = df.index.to_numpy()
-        count = df["count"].to_numpy()
-        all_counts = np.array([])
-        for label in all_labels:
-            index = np.where(labels == label)
-            if len(index[0]) == 0:
-                all_counts = np.append(all_counts, 0)
-            else:
-                all_counts = np.append(all_counts, count[index[0]])
-        if show_diagram:
-            plt.bar(all_labels, all_counts)
-            plt.title("Histogram for top {} points per label".format(k))
-            plt.yticks(np.arange(0, np.int_(max(all_counts)) + 1, 1))
-            plt.show()
-        return all_labels, all_counts
+        return self._to_json()
 
-    def histogram_top_k_relative(
-        self,
-        all_labels: np.ndarray,
-        k: int = 10,
-        show_diagram: bool = True,
-    ) -> Tuple[np.ndarray, np.ndarray]:
+    def _convertToListIfNeeded(self, obj):
```

Comment:
Can you please rename this to camel case style (e.g. `_convert_to_list_if_needed`)?

Reply:
done